lib/node/nodes/aggregate-intersection.js

```diff
@@ -12,34 +12,46 @@ var PARAMS = {
 
 var AggregateIntersection = Node.create(TYPE, PARAMS, { cache: true, version: 3 });
 
+var ColumnNamer = require('../column-namer');
+
+
 module.exports = AggregateIntersection;
 
+
 AggregateIntersection.prototype.sql = function() {
     var groupByColumns = setAliasPrefixToColumnNames(this.source.getColumns(), '_cdb_analysis_source');
     var columns = setAliasPrefixToColumnNames(this.source.getColumns(), '_cdb_analysis_source');
     var finalColumnName = this.aggregate_function;
     var qualifiedAggregateColumn = '_cdb_analysis_target.';
+    var columnNamer = new ColumnNamer(columns);
+
     if (this.aggregate_column === null) {
         qualifiedAggregateColumn += '*';
     } else {
         qualifiedAggregateColumn += this.aggregate_column;
         finalColumnName += '_' + this.aggregate_column;
     }
 
+    if (this.aggregate_function === 'count') {
```

Comment:
Can we extract everything related to assign a unique name to the output column to a its own method?

Reply:
That line isn't really related to the column renaming, it copes with the different name schema used for `count` aggregations. I moved that bit from where it was to be able to adjust the column name after it was computed.

So we already have all the column-ranaming stuff in this single line:
```
finalColumnName = columnNamer.uniqueName(finalColumnName);
```